np.Templating/src/Templating.js

```diff
@@ -273,6 +273,116 @@ export async function templateNew(): Promise<void> {
+ * Note: writeTypeType === 'prepend' prepends, otherwise appends
+ * Note: writeType will be 'append' or 'prepend' | if writeUnderHeading is set, then appends/prepends there, otherwise the note's content
+ * Note: if you are inserting title text as part of your template, then you should always prepend, because your title will confuse future appends
+ * - .addParagraphBelowHeadingTitle(title, paragraphType, headingTitle, shouldAppend: Bool, shouldCreate)
+ * Note: ask CD what the reserved frontmatter fields should be and trap for them
+ * xcallback note: arg1 is template name, arg2 is whether to open in editor, arg3 is a list of vars to pass to template equals sign is %3d
+ *
+ */
+export async function templateFileByTitle(selectedTemplate?: string, openInEditor?: boolean = false, args?: string = ''): Promise<void> {
+  try {
+    let argObj = {}
+    args.split(',').forEach((arg) => (arg.split('=').length === 2 ? (argObj[arg.split('=')[0]] = arg.split('=')[1]) : null))
+    if (!selectedTemplate || selectedTemplate.length === 0) {
+      await CommandBar.prompt(`templateFileByTitle: no templateTitle was specified."`, helpInfo('Presets'))
+    }
+    let failed = false
+    const templateData = await NPTemplating.getTemplate(selectedTemplate)
+    if (!templateData) {
+      failed = true
+    }
+    const isFrontmatter = new FrontmatterModule().isFrontmatterTemplate(templateData)
+    if (!failed && isFrontmatter) {
+      const templateAttributes = await NPTemplating.getTemplateAttributes(templateData)
+      const { frontmatterBody, frontmatterAttributes } = await NPTemplating.preRender(templateData)
+      // Note: frontmatterAttributes and templateAttributes are the same
+      let data = { ...frontmatterAttributes, ...argObj, frontmatter: { ...frontmatterAttributes, ...argObj } }
+      let renderedTemplate = await NPTemplating.render(frontmatterBody, data)
+
+      const { openNoteTitle, writeNoteTitle, writeType, writeUnderHeading } = frontmatterAttributes
+      let noteTitle = (openNoteTitle && openNoteTitle.trim()) || (writeNoteTitle && writeNoteTitle?.trim()) || ''
+      let shouldOpenInEditor = openNoteTitle && openNoteTitle.length > 0
+      const shouldAppend = (writeType && writeType === 'append') || false
+      const isTodayNote = /<today>/i.test(openNoteTitle)
+      let note
+      if (isTodayNote) {
+        await Editor.openNoteByDate(new Date(), false)
+        await writeNoteContents(Editor.note, renderedTemplate, writeUnderHeading, shouldAppend)
+      } else {
+        if (noteTitle.length) {
+          const notes = await DataStore.projectNoteByTitle(noteTitle)
+          if (!notes || notes.length == 0 || (notes && notes.length > 1)) {
+            const msg = `${!notes || notes.length == 0 ? 'no' : notes.length > 1 ? 'more than one' : ''}`
+            await CommandBar.prompt(`"${noteTitle}" matches ${msg} note(s) title. the title needs to be distinct to ensure correct note is written to.`, helpInfo('Presets'))
```

Comment:
Rephase to use `unique` instead of `distinct` more "human", less "techy" :-)

Reply:
agreed. done.